internal/settings/providers/porkbun/provider.go

```diff
@@ -0,0 +1,295 @@
+		apiKey:       extraSettings.ApiKey,
+		ttl:          extraSettings.TTL,
+		logger:       logger,
+	}
+	if err := p.isValid(); err != nil {
+		return nil, err
+	}
+	return p, nil
+}
+
+func (p *provider) isValid() error {
+	switch {
+	case len(p.apiKey) == 0:
+		return errors.ErrEmptyAppKey
+	case len(p.secretApiKey) == 0:
+		return errors.ErrEmptyConsumerKey
+	}
+	return nil
+}
+
+func (p *provider) String() string {
+	return fmt.Sprintf("[domain: %s | host: %s | provider: Porkbun]", p.domain, p.host)
+}
+
+func (p *provider) Domain() string {
+	return p.domain
+}
+
+func (p *provider) Host() string {
+	return p.host
+}
+
+func (p *provider) IPVersion() ipversion.IPVersion {
+	return p.ipVersion
+}
+
+func (p *provider) Proxied() bool {
+	return false
+}
+
+func (p *provider) BuildDomainName() string {
+	return utils.BuildDomainName(p.host, p.domain)
+}
+
+func (p *provider) HTML() models.HTMLRow {
+	return models.HTMLRow{
+		Domain:    models.HTML(fmt.Sprintf("<a href=\"http://%s\">%s</a>", p.BuildDomainName(), p.BuildDomainName())),
+		Host:      models.HTML(p.Host()),
+		Provider:  "<a href=\"https://www.porkbun.com/\">Porkbun DNS</a>",
+		IPVersion: models.HTML(p.ipVersion.String()),
+	}
+}
+
+func (p *provider) setHeaders(request *http.Request) {
+	headers.SetUserAgent(request)
+	headers.SetContentType(request, "application/json")
+	headers.SetAccept(request, "application/json")
+}
+
+func (p *provider) getRecords(ctx context.Context, client *http.Client) (recordIDs []string, err error) {
+	u := url.URL{
+		Scheme: "https",
+		Host:   "porkbun.com",
+		Path:   fmt.Sprintf("/api/json/v3/dns/retrieve/%s", p.domain),
+	}
+	postRecordsParams := struct {
+		SecretApiKey string `json:"secretapikey"`
+		ApiKey       string `json:"apikey"`
+	}{
+		SecretApiKey: p.secretApiKey,
+		ApiKey:       p.apiKey,
+	}
+	bodyBytes, err := json.Marshal(postRecordsParams)
+	if err != nil {
+		return nil, fmt.Errorf("%w: %s", errors.ErrRequestMarshal, err)
+	}
+
+	p.logger.Debug("HTTP POST getRecords: " + u.String() + ": " + string(bodyBytes))
+
+	request, err := http.NewRequestWithContext(ctx, http.MethodPost, u.String(), bytes.NewBuffer(bodyBytes))
+	if err != nil {
+		return nil, err
+	}
+	p.setHeaders(request)
+
+	response, err := client.Do(request)
+	if err != nil {
+		return nil, fmt.Errorf("%w: %s", errors.ErrUnsuccessfulResponse, err)
+	}
+
+	if response.StatusCode != http.StatusOK {
+		return nil, fmt.Errorf("%w: %d: %s",
+			errors.ErrBadHTTPStatus, response.StatusCode, utils.BodyToSingleLine(response.Body))
+	}
+
+	type domainRecords struct {
+		Status  string `json:"status"`
+		Records []struct {
+			Id      string `json:"id"`
+			Name    string `json:"title"`
+			Type    string `json:"type"`
+			Content string `json:"content"`
+			TTL     string `json:"ttl"`
+			Prio    string `json:"prio"`
+			Notes   string `json:"notes"`
+		} `json:"records"`
+	}
+	defer response.Body.Close()
+	var responseData domainRecords
```

Comment:
- inline type
- remove `defer response.Body.Close()` in favor of moving it up
- Trim down to required fields for the record struct

```suggestion
	var responseData struct {
		Records []struct {
			Id      string `json:"id"`
			Content string `json:"content"`
		} `json:"records"`
	}
```